src/vector/index.js

```diff
@@ -428,4 +446,63 @@ async function loadLanguage() {
     }
 }
 
+async function verifyServerConfig() {
+    console.log("Verifying homeserver configuration");
+
+    // TODO: Handle query string arguments for hs_url and is_url
+    // We probably don't want to handle them unless the user is logged out though?
+
+    const config = SdkConfig.get();
+    let wkConfig = config['default_server_config']; // overwritten later under some conditions
+    const serverName = config['default_server_name'];
+    const hsUrl = config['default_hs_url'];
+    const isUrl = config['default_is_url'];
+
+    const incompatibleOptions = [wkConfig, serverName, hsUrl].filter(i => !!i);
+    if (incompatibleOptions.length > 1) {
+        throw newTranslatableError(_td(
+            "Invalid configuration: can only specify one of default_server_config, default_server_name, " +
+            "or default_hs_url.",
+        ));
+    }
+
+    if (hsUrl) {
+        console.log("Config uses a default_hs_url - constructing a default_server_config using this information");
+
+        wkConfig = {
+            "m.homeserver": {
+                "base_url": hsUrl,
+            },
+        };
+        if (isUrl) {
+            wkConfig["m.identity_server"] = {
+                "base_url": isUrl,
+            };
+        }
+    }
+
+    let result = null;
+
+    if (wkConfig) {
+        console.log("Config uses a default_server_config - validating object");
+        result = await AutoDiscovery.fromDiscoveryConfig(wkConfig);
+    }
+
+    if (serverName) {
+        console.log("Config uses a default_server_name - doing .well-known lookup");
+        result = await AutoDiscovery.findClientConfig(serverName);
+    }
+
+    const validatedConfig = AutoDiscoveryUtils.buildValidatedConfigFromDiscovery(serverName, result);
+
+    // Just in case we ever have to debug this
+    console.log("Using homeserver config:", validatedConfig);
+
+    // Add the newly built config to the actual config for use by the app
+    console.log("Updating SdkConfig with validated discovery information");
+    SdkConfig.add({"validated_discovery_config": validatedConfig});
```

Comment:
discovery is maybe not very explanatory, maybe something like `validated_server_config`?